modules/kafka/kafka-dest-worker.c

```diff
@@ -150,12 +157,23 @@ kafka_dest_worker_free(LogThreadedDestWorker *s)
   log_threaded_dest_worker_free_method(s);
 }
 
+static gboolean
+_poller_thread_init(LogThreadedDestWorker *s)
+{
+  KafkaDestDriver *owner = (KafkaDestDriver *) s->owner;
+  s->time_reopen = owner->poll_timeout / 1000;
+
+  return log_threaded_dest_worker_init_method(s);
+}
+
 LogThreadedDestWorker *
 kafka_dest_worker_new(LogThreadedDestDriver *o, gint worker_index)
 {
   KafkaDestWorker *self = g_new0(KafkaDestWorker, 1);
 
   log_threaded_dest_worker_init_instance(&self->super, o, worker_index);
+  if (_is_poller_thread(self))
+    self->super.thread_init = _poller_thread_init;
```

Comment:
I would just initialize the function pointer to a static value here and branch on the same conditional there. It doesn't have performance implication but makes the code easier to follow.

Reply:
It's not about performance. My intention was to express that in the other case we don't have to implement the original method.
I'll change it (and will implement another method for the non-poller case that will just call the original method - and nothing else).

Reply:
finally I just merged the two in one `_thread_init`